lib/active_merchant/billing/gateways/cams.rb

```diff
@@ -0,0 +1,241 @@
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+
+        commit("sale", post)
+      end
+
+      def authorize(money, payment, options={})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+
+        commit('auth', post)
+      end
+
+      def capture(money, authorization, options={})
+        transaction_id, authcode = split_authorization(authorization)
+        post = {
+          "transactionid" => transaction_id,
+          "authcode"      => authcode
+        }
+        add_invoice(post, money, options)
+        add_customer_data(post, options)
+
+        commit('capture', post)
+      end
+
+      def refund(money, authorization, options={})
+        transaction_id, authcode = split_authorization(authorization)
+        post = {
+          "transactionid" => transaction_id,
+          "authcode"      => authcode
+        }
+        add_invoice(post, money, options)
+        commit('refund', post)
+      end
+
+      def void(authorization, options={})
+        transaction_id, authcode = split_authorization(authorization)
+        post = {
+          "transactionid" => transaction_id,
+          "authcode"      => authcode
+        }
+        commit('void', post)
+      end
+
+      #def verify(credit_card, options={})
+      #  MultiResponse.run(:use_first_response) do |r|
+      #    r.process { authorize(1 + rand(50), credit_card, options) }
+      #    r.process(:ignore_result) { void(r.authorization, options) }
+      #  end
+      #end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        [:ccnumber, :ccexp, :cvv, :address1, :address2, 
+        :city, :password, :password,
+        :firstname, :lastname, :authcode, :state,
+        :phone, :zip
+        ].each do |key|
+          transcript = transcript.
+            gsub(%r((#{key}=)[^&]+), '\1[FILTERED]\2')
+        end
+
+        transcript
+      end
+
+      private
+
+      def add_customer_data(post, options)
+      end
+
+      def add_address(post, creditcard, options)
+        address = options[:billing_address]
+        post.update({
+          :firstname => creditcard.first_name,
+          :lastname  => creditcard.last_name,
+          :address1  => address[:address1],
+          :address2  => address[:address2],
+          :city      => address[:city],
+          :state     => address[:state],
+          :zip       => address[:zip],
+          :country   => "US", #address[:country],
+          :phone     => address[:phone]
+        })
+      end
+
+      def add_invoice(post, money, options)
+        post[:amount] = amount(money)
+        post[:currency] = (options[:currency] || currency(money))
+      end
+
+      def add_payment(post, payment)
+        post.update({
```

Comment:
Prefer the `post[:ccnumber] = payment.number` style, rather than using `update`.